nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -791,7 +789,8 @@ private Description checkReturnExpression(
     Type returnType = methodSymbol.getReturnType();
     if (returnType.isPrimitiveOrVoid()) {
       // check for unboxing
-      return returnType.isPrimitive() ? doUnboxingCheck(state, retExpr) : Description.NO_MATCH;
+      doUnboxingCheck(state, retExpr);
```

Comment:
What happens with unboxing of void here? Does it still work or does a `null` `retExpr` trip it up? (I presume we have a test case for this somewhere, but I am not 100% sure...)

Reply:
Good catch! I will make sure there is a test for this. But I think the unboxing check will only report an error if the expression has a primitive type (void doesn't count). Will check it

Reply:
Ok, I checked this.  We do have tests for returning `null` with a `Void` return type:

https://github.com/uber/NullAway/blob/d4f4d89cac20e1557a94247c3150e65b782c805d/nullaway/src/test/java/com/uber/nullaway/NullAwayCoreTests.java#L671-L671

But this code is not relevant for that; `Void` is handled by the next condition in the function:

https://github.com/uber/NullAway/blob/d4f4d89cac20e1557a94247c3150e65b782c805d/nullaway/src/main/java/com/uber/nullaway/NullAway.java#L795-L800

`isPrimitiveOrVoid()` actually checks for either a primitive or (lowercase) `void` type.  I think lowercase `void` should be impossible here, though, since we have a `retExpr` that cannot be `null` (assuming our NullAway checking is thorough enough).  Probably calling `isPrimitiveOrVoid()` was a previous misunderstanding; I've changed the call to just use `isPrimitive()` and all tests still pass.